lib/routes/maze-routes.js

```diff
@@ -23,9 +23,20 @@ router
       delete query.number;
     }
 
-    Maze.find(query)
-      .limit(number)
-      .lean()
+    const convertToNumber = (obj, key) => {
+      if(obj[key]) obj[key] = Number(obj[key]);
+    };
+    convertToNumber(query, 'dimensions.width');
+    convertToNumber(query, 'dimensions.height');
+    if(query.solutionLength) convertToNumber(query.solutionLength, '$lt');
+    if(query.solutionLength) convertToNumber(query.solutionLength, '$gt');
+    if(query.connectivity) convertToNumber(query.connectivity, '$lt');
+    if(query.connectivity) convertToNumber(query.connectivity, '$gt');
+    
```

Comment:
There's also averagePathLength, and the functionality supports greater than/ less than or equal to. So I think we need to add the following:

`    
if(query.solutionLength) convertToNumber(query.solutionLength, '$lte');
if(query.solutionLength) convertToNumber(query.solutionLength, '$gte');
if(query.connectivity) convertToNumber(query.connectivity, '$lte');
if(query.connectivity) convertToNumber(query.connectivity, '$gte');
if(query.averagePathLength) convertToNumber(query.averagePathLength, '$lt');
if(query.averagePathLength) convertToNumber(query.averagePathLength, '$gt');`
if(query.averagePathLength) convertToNumber(query.averagePathLength, '$lte');
if(query.averagePathLength) convertToNumber(query.averagePathLength, '$gte');
`

Reply:
This all looks good to me. I would agree with the additions that sam added.

Reply:
Upon further review:

- Lines 29 and 30 appear to assume that there will necessarily be a height and a width specified. I don't think that's the case.
- I just remembered that height and width also have lt, gt, lte, and gte functionality as well.

And lastly, it appears that on lines 31-34 we're checking whether those keys exist in query once, and then running convertToNumber and checking again. Unless I'm missing something, that seems redundant and could probably be removed